app/views/recommended_feeds/show.js.erb

```diff
@@ -0,0 +1,8 @@
+$(function() {
+  var element = $(".<%= recommended_feed_item_container_css_class(@feed_item.id) %>");
+  <% if @feed_item.recommended_by_user? %>
+    element.replaceWith("<%= j (render @feed_item) %>");
+  <% else %>
+    element.remove();
+  <% end %>
+})
```

Comment:
Ajax-enabled for a better user experience. No postbacks when UNrecommending (UNstarring) a feed item. This javascript handles updating the UI for recommended (starred) items. If a feed item is UNrecommended and no other users have recommendations outstanding on that feed item, it is removed from the UI without a postback. If there ARE additional recommendations, the item is UPDATED to reflect this in the UI, but is not removed.